### PR TITLE
KT-17379: Converting multiline expressions creates dangling operations fixed

### PR DESCRIPTION
**Pull request by Dimach**

### CODE REVIEW

File: compiler/frontend/src/org/jetbrains/kotlin/psi/KtPsiUtil.java [hunk 1]
```diff
@@ -513,6 +513,16 @@ public static boolean areParenthesesNecessary(
             return false;
         }
 
+        if (innerExpression instanceof KtBinaryExpression) {
+            PsiElement expr = innerExpression.getFirstChild();
+            while (expr != null) {
+                if (expr instanceof PsiWhiteSpace && (expr.textContains('\n') || expr.textContains('\r'))) {
```
Comment (yole) [inline compiler/frontend/src/org/jetbrains/kotlin/psi/KtPsiUtil.java]: IntelliJ always normalizes text of files to use \n line breaks, the \r will never be there. Please remove it from the check.

File: j2k/testData/fileOrElement/kotlinApiAccess/ObjectMembers.kt [hunk 2]
```diff
@@ -4,8 +4,8 @@ internal class C {
     fun foo(): Int {
         KotlinObject.property1 = 1
         KotlinObject.property2 = 2
-        return KotlinObject.foo() +
+        return (KotlinObject.foo() +
```
Comment (yole) [inline j2k/testData/fileOrElement/kotlinApiAccess/ObjectMembers.kt]: The parentheses are only required if the line break is before the operator. In this case they should not be added.

**Comment by Dimach** [comment]
Done.

File: j2k/testData/fileOrElement/kotlinApiAccess/CorrectNullabilityDetected.kt [hunk 3]
```diff
@@ -3,13 +3,13 @@ import kotlinApi.*
 //TODO: Formatter works incorrectly
```
Comment (piotrek1543) [inline j2k/testData/fileOrElement/kotlinApiAccess/CorrectNullabilityDetected.kt]: After changes this `TODO` should be removed.

File: j2k/testData/fileOrElement/kotlinApiAccess/CorrectNullabilityDetected.kt [hunk 4]
```diff
@@ -3,13 +3,13 @@ import kotlinApi.*
...
     fun foo(c: KotlinClass): Int {
-        return c.nullableProperty!!.length
-        +c.property.length
-        +KotlinClass.nullableStaticVar!!
-        +KotlinClass.staticVar
-        +KotlinClass.nullableStaticFun(1)!!
-        +KotlinClass.staticFun(1)
-        +nullableGlobalFunction("")!!.length
-        +globalFunction("").length
+        return (c.nullableProperty!!.length
```
Comment (piotrek1543) [inline j2k/testData/fileOrElement/kotlinApiAccess/CorrectNullabilityDetected.kt]: After changes `TODO` comment should be removed

File: j2k/testData/fileOrElement/strings/formattedString.kt [hunk 5]
```diff
@@ -4,7 +4,7 @@ object A {
             "text3"
 
 
-    val TEXT2 = "text1\n"
-    + "text2\n"
-    + "text3"
+    val TEXT2 = ("text1\n"
```
Comment (piotrek1543) [inline j2k/testData/fileOrElement/strings/formattedString.kt]: If I were you, I would reduce space from 2 to 1 line. I think one is enough to make this object readable.

File: j2k/testData/fileOrElement/kotlinApiAccess/CorrectNullabilityDetected.kt (hunk 4 above)
Comment (piotrek1543) [inline j2k/testData/fileOrElement/kotlinApiAccess/CorrectNullabilityDetected.kt]: After changes TODO comment above your code should be removed. You've already fixed it.

File: j2k/testData/fileOrElement/strings/formattedString.kt (hunk 5 above)
Comment (Dimach) [inline j2k/testData/fileOrElement/strings/formattedString.kt]: You write comment to proper file? I dont see any problem here. Maybe you mean kt-17379.kt?

**Comment by Dimach** [comment]
I improved new lines recognition (diff you can see in orWithNewLine.kt) but i not fix problem with "end of line comments" that produce spare newlines in this PR, its different task as i think.

**Comment by semoro** [comment]
Merged manually. 
Thanks! 🙂

**Pull request closed without merging**